### PR TITLE
Normalize SerialManagers ports defaults to allow inclusion in hwdefs

### PR DESCRIPTION
**Pull request by Hwurzburg**
Needed to move serial params out of defaults.parm and into hwdefs. Makes things consistent and follows new pattern for defaults...

@peterbarker many of the existing baud rate defines are wrong...would like to discuss with you how we should change them...I think they should match the param input value....not divided by 1000 and then truncated by a implied cast....and many just don't work anyway now I think....

### CODE REVIEW

File: libraries/AP_SerialManager/AP_SerialManager.cpp [hunk 1]
```diff
@@ -98,26 +124,26 @@ extern const AP_HAL::HAL& hal;
...
+#define DEFAULT_SERIAL0_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL1_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL2_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL3_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL4_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL5_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL6_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL7_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL8_PROTOCOL SerialProtocol_None
+#define DEFAULT_SERIAL9_PROTOCOL SerialProtocol_None
```
Comment (peterbarker) [inline libraries/AP_SerialManager/AP_SerialManager.cpp:146]: I'll probably move this crap down into `chibios_hwdef.py` where we set the defaults for AP_Periph

**Review by peterbarker — COMMENTED** [review]
Should be fix the default of "0_BAUD" while we're at it?

File: libraries/AP_SerialManager/AP_SerialManager.h [hunk 2]
```diff
@@ -51,11 +51,8 @@
 
 
  // console default baud rates and buffer sizes
-#ifdef HAL_SERIAL0_BAUD_DEFAULT
-# define AP_SERIALMANAGER_CONSOLE_BAUD          HAL_SERIAL0_BAUD_DEFAULT
-#else
+
 # define AP_SERIALMANAGER_CONSOLE_BAUD          115200
```
Comment (tridge) [inline libraries/AP_SerialManager/AP_SerialManager.h:55]: can we remove the CONSOLE defines now?

Comment (Hwurzburg) [reply libraries/AP_SerialManager/AP_SerialManager.h:55]: it is used in AP_Periph and ESP32 also, as well as the default for serial 0

File: libraries/AP_HAL_ChibiOS/hwdef/BirdCANdy/hwdef.dat [hunk 3]
```diff
@@ -45,7 +45,7 @@ PB5  TIM3_CH2  TIM3 PWM(4) GPIO(53)
 # USART1 for debug
 # PA15 USART1_TX USART1
 # PB3 USART1_RX USART1
-# define HAL_SERIAL0_BAUD_DEFAULT 57600
+# define DEFAULT_SERIAL0_BAUD_DEFAULT 57600
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/BirdCANdy/hwdef.dat:48]:
```suggestion
# define DEFAULT_SERIAL0_BAUD 57600
```

similarly elsewhere

File: libraries/AP_HAL_ChibiOS/hwdef/CubeRedSecondary/hwdef.dat [hunk 4]
```diff
@@ -146,10 +146,10 @@ PD3 USART2_CTS USART2 SPEED_HIGH
 SERIAL_ORDER UART7 UART8 USART3 USART6 UART4 USART2 
 
 # use 2 MBaud when talking to primary controller
-define HAL_SERIAL0_BAUD_DEFAULT 2000000
-define HAL_SERIAL2_PROTOCOL SerialProtocol_RCIN
-define HAL_SERIAL3_PROTOCOL SerialProtocol_Sbus1
-define HAL_SERIAL4_PROTOCOL SerialProtocol_RCIN
+define DEFAULT_SERIAL0_BAUD_DEFAULT 2000000
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/CubeRedSecondary/hwdef.dat:149]:
```suggestion
define DEFAULT_SERIAL0_BAUD 2000000
```

File: libraries/AP_HAL_ChibiOS/hwdef/Sierra-F412/hwdef.dat [hunk 5]
```diff
@@ -40,7 +40,7 @@ SERIAL_ORDER USART1 EMPTY EMPTY USART2
 # USART1 for debug
 PA9 USART1_TX USART1
 PA10 USART1_RX USART1
-define HAL_SERIAL0_BAUD_DEFAULT 57600
+define DEFAULT_SERIAL0_BAUD_DEFAULT 57600
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/Sierra-F412/hwdef.dat:43]:
```suggestion
define DEFAULT_SERIAL0_BAUD 57600
```

... more of these around...

**Review by peterbarker — CHANGES_REQUESTED** [review]
This PR should be an NFC.  However, even after fixing Sierra-F412 it is not - `size_compare_branches.py` says it is not output-identical.

```
Board,AP_Periph
Sierra-F412,0
```

That's 0 byte change but not an identical binary.  Using objdump on the two branches gives this as part of its output:
```
- 801a88a:	f44f 4161 	mov.w	r1, #57600	; 0xe100
+ 801a88a:	f44f 31e1 	mov.w	r1, #115200	; 0x1c200
```

so somewhere in this PR we've changed a default on this board from 57600 to 115200

Use
```
./Tools/scripts/size_compare_branches.py --board=Sierra-F412 --all-vehicles
```
to recreate

File: libraries/AP_HAL_ChibiOS/hwdef/BirdCANdy/hwdef.dat (hunk 3 above)
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/BirdCANdy/hwdef.dat]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/CubeRedSecondary/hwdef.dat (hunk 4 above)
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/CubeRedSecondary/hwdef.dat]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/Sierra-F412/hwdef.dat (hunk 5 above)
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/Sierra-F412/hwdef.dat]: fixed the rest

**Comment by Hwurzburg** [comment]
> 

well I cant find it....I did not mess with AP_Periph and a grep for HAL_SERIAL0 yields nothing

**Comment by peterbarker** [comment]
```
Peter Barker: I've checked that PR now, and I'm happy to merge the serial one post my fixes.
[6:51 PM]Peter Barker: I've built all of the boards and I'm satisfied that I can explain all ofthe changes made in that PR.  Am I OK to force-push over your branch with my branch?
[8:48 PM]Henry Wurzburg: Sure
...thanks (edited)

1
[8:52 PM]Peter Barker: I've done that forced-push now.
[8:53 PM]Peter Barker: The periphs are the ones we can be sure about as they don't include the hwdef.dat in the build product.
[8:54 PM]Peter Barker: All of the other boards which didn't have identical output had lines matching SERIAL_PROTOCOL.* in their hwdef files, and all size changes were of the sort that could explained by changes in the romfs.
[8:55 PM]Peter Barker: I spot-checked a couple using elf-diff and that showed only changes in the romfs symbol.
[8:55 PM]Peter Barker: .... actually, exluding ROMFS from the build would actually be a good thing to add to size_compare_branches.py, come to think of it.
```

**Comment by Hwurzburg** [comment]
thanks.... @peterbarker

**Comment by peterbarker** [comment]
@Hwurzburg needs a rebase.  Have you finished looking through this now?

**Review by peterbarker — APPROVED** [review]
LGTM

**Pull request merged**